utreexo-validation-bip.md

```diff
@@ -0,0 +1,328 @@
+we need to commit to this value to avoid malicious peers that may send over the
+wrong amount.
+
+##### script pubkey size
+
+As the script pubkey is a variable length byte array, we prepend it with the
+length.
+
+##### script pubkey
+
+This field is added to commit to the locking script of the UTXO. With current
+nodes, this is stored in the UTXO set but since we receive this in the proof
+from our peers, we need to commit to this value to avoid malicious peers that
+may send over the wrong locking script.
+
+#### Hash function
+
+The leaf data is hashed with SHA-512/256, which gives us a 32 byte hash.
+It was chosen over SHA-256 due to the faster performance on 64 bit systems.
+
+#### Excluded UTXOs from the accumulator
+
+Not all transaction outputs are added to a node's UTXO set. Normal Bitcoin nodes
+only form consensus on the set of transactions, not on the UTXO set, so different
+nodes can omit different outputs and stay compatible as long as those outputs are
+never spent. Utreexo nodes, however, do require explicit consensus on the UTXO set
+as all proofs are with respect to the Merkle roots of the entire set.
+
+For this reason, we define which UTXOs are not inserted to the accumulator.  Any
+variations here will result in Utreexo nodes with incompatible proofs.
+
+##### Provably unspendable transaction outputs
+
+There are outputs in the Bitcoin network that we can guarantee that they cannot
+be spent without a hard-fork of the network. The following output types are not
+added to the accumulator:
+- Outputs that start with an OP_RETURN (0x6a)
+- Outputs with a scriptPubkey larger than 10,000 bytes
+
+##### Same block spends
+
+Often, UTXOs are created and spent in the same block. This is allowed by Bitcoin
+consensus rules as long as the output being spent is created by a transaction earlier
+in the block than the spending transaction.
+In Utreexo, nodes inspect blocks and identify which outputs are being created
+and destroyed in the same block, and exclude them from the accumulator and proofs.
+
+There's no need to provide proofs for outputs which have been created in the same
+block. Adding and then immediately removing the output from the accumulator would be
+possible but doesn't serve any purpose - once outputs are spent their past existence
+cannot be proven with the Utreexo accumulator (and SPV proofs already provide that).
+
+For these reasons, outputs which are spent in the same block where they are created
+are omitted from the accumulator, and those inputs are omitted from block proofs.
+
+#### Order of operations
+
+The Utreexo accumulator lacks associative properties during addition and the
+ordering of which UTXO hash gets added first is consensus critical. For
+the modification of the accumulator the steps are as follows:
+
+1. Batch remove the UTXOs that were spent in the block based on the algorithm
+   defined in BIP-????. Deletions itself are order-independent.
```

Comment:
For the time being, maybe just use `[‎BIP Utreexo Accumulator](‎utreexo-accumulator-bip.md)` to clarify whether you are referring to one or the other.